presto-manager-common/src/main/java/com/teradata/prestomanager/common/ApiRequester.java

```diff
@@ -41,26 +41,42 @@
 
     private Client client;
 
-    private ApiRequester(Client client, // TODO: Use AssistedInject to manage this
+    private ApiRequester(Client client,
             UriBuilder uriBuilder, HttpMethod method,
             Entity entity, MultivaluedMap<String, Object> headers,
             String mediaType)
     {
         this.client = requireNonNull(client);
-        this.uriBuilder = requireNonNull(uriBuilder);
+        this.uriTemplate = requireNonNull(uriBuilder).toTemplate();
         this.method = requireNonNull(method);
         this.headers = requireNonNull(headers);
         this.mediaType = requireNonNull(mediaType);
 
         this.entity = entity;
     }
 
+    /**
+     * Send this request to the given base URI.
+     * <p>
+     * The path should contain no components besides user-info, host, and port.
+     * <p>
+     * This method is thread-safe.
+     */
+    // TODO: The uri should be validated to not conflict with the uriTemplate
+    // If the uriTemplate contains a path, and the given URI has a path (even
+    // just a single trailing slash, the template's path will be replaced).
     public Response send(URI uri)
```

Comment:
I've annotated `ApiRequester` and `RequestDispatcher` as `ThreadSafe` (the annotation only applies to classes).

Maybe I should annotate `ThreadSafe` with the more-specific `Immutable` annotation instead? Or use both?